packages/module/src/ColumnManagementModal/ColumnManagementModal.tsx

```diff
@@ -0,0 +1,146 @@
+import React from 'react';
+import {
+  Modal,
+  Button,
+  TextContent,
+  Text,
+  TextVariants,
+  DataListItem,
+  DataList,
+  DataListItemRow,
+  DataListCheck,
+  DataListCell,
+  DataListItemCells,
+  Split,
+  SplitItem,
+  ModalProps
+} from '@patternfly/react-core';
+
+export interface ColumnManagementModalColumn {
+  /** Internal identifier of a column by which table displayed columns are filtered. */
+  key: string,
+  /** The actual display name of the column possibly with a tooltip or icon. */
+  title: React.ReactNode,
+  /** If user changes checkboxes, the component will send back column array with this property altered. */
+  isShown?: boolean,
+  /** Set to false if the column should be hidden initially */
+  isShownByDefault: boolean,
+  /** The checkbox will be checked and disabled, this is applicable to identifier columns which user shouldn't be able to hide */
+  isAlwaysShown?: boolean
```

Comment:
Correct me if I am wrong, but from the code, this seems to only disable the checkbox, not ensure it is checked. Perhaps we could rename it to `isDisabled` or change the logic so that `isShown = true` doesn't have to be present

Reply:
Yes, you are right; `isDisabled` in the context of column config could be understood as the disabling of the whole column, so I went with `isUntoggleable`, it's a bit of a mouthful, but I didn't want to go with `isToggleable` so that the default value would not be `true`. If you have better name for the param, I will gladly go with it.

Reply:
That's a fair point, thank you!